File: Backend/ProjectDashboard/Models/Domain/ProjectTask.cs
namespace ProjectDashboard.Models.Domain {
	public class ProjectTask {

		public int Id { get; set; }

		public string Title { get; set; }

		public string Description { get; set; }

		public bool IsFinished { get; set; } = false;

		public int ProjectId { get; set; }

		public Project Project { get; set; }


		public int? AssigneeId { get; set; }

		public User Assignee { get; set; }

		public ProjectTask() { }

		public ProjectTask(string title, string description, Project project, User assignee = null) {
			this.Title = title;
			this.Description = description;
			this.Project = project;
			this.Assignee = assignee;
		}
	}
}

File: Android/app/src/main/java/com/quickdev/projectdashboard/models/domain/ProjectTask.kt
package com.quickdev.projectdashboard.models.domain

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "tasks",
    indices = [
        Index(value = ["id"], unique = true)
    ]
)
data class ProjectTask(
    @PrimaryKey(autoGenerate = false)
    val id: Int = 0,
    var title: String,
    val projectId: Int,
    var description: String,
    var isFinished: Boolean = false,
    var assigneeId: Int?
) {
    @Ignore
    var assignee: User? = null
}
